encfs/CipherFileIO.cpp

```diff
@@ -168,18 +168,33 @@ int CipherFileIO::getAttr(struct stat *stbuf) const {
   // stat() the backing file
   int res = base->getAttr(stbuf);
 
```

Comment:
The change adds a few cases to the "if maze". Please simplify the whole thing a bit by doing

    if(res != 0)
        return res;
    if(!S_ISREG(stbuf->st_mode))
       // special files don't need size adjustment.
       return 0;
    if(stbuf->st_size == 0)
        // empty files stay empty
       return 0;

here. In other words, handle the simple cases first and early return.